File: server/models/cart_product.js
const { Schema, model } = require('mongoose');

const cartProductSchema = Schema({
  product: { type: Schema.Types.ObjectId, ref: 'Product' },
  quantity: { type: Number, default: 1 },
  selectedSize: String,
  selectedColour: String,
  productName: { type: String, required: true },
  productImage: { type: String, required: true },
  productPrice: { type: Number, required: true },
  reservationExpiry: {
    type: Date,
    default: () => new Date(Date.now() + 30 * 60 * 1000),
  },
  reserved: { type: Boolean, deafult: true },
});

cartProductSchema.set('toJSON', { virtuals: true });
cartProductSchema.set('toObject', { virtuals: true });

exports.CartProduct = model('CartProduct', cartProductSchema);

File: Mobile/lib/src/cart/domain/entities/cart_product.dart
import 'dart:ui';

import 'package:equatable/equatable.dart';

class CartProduct extends Equatable {
  const CartProduct({
    required this.id,
    required this.productId,
    required this.quantity,
    required this.productName,
    required this.productImage,
    required this.productPrice,
    this.selectedSize,
    this.selectedColour,
    required this.productExists,
    required this.productOutOfStock,
  });

  const CartProduct.empty()
      : id = "Test String",
        productId = "Test String",
        quantity = 1,
        productName = "Test String",
        productImage = "Test String",
        productPrice = 1,
        selectedSize = null,
        selectedColour = null,
        productExists = true,
        productOutOfStock = true;

  final String id;
  final String productId;
  final int quantity;
  final String productName;
  final String productImage;
  final double productPrice;
  final String? selectedSize;
  final Color? selectedColour;
  final bool productExists;
  final bool productOutOfStock;

  @override
  List<dynamic> get props => [
        id,
        productId,
        quantity,
        productName,
        productImage,
        productPrice,
        selectedSize,
        selectedColour,
        productExists,
        productOutOfStock,
      ];
}
